test/models/playlist_test.rb

```diff
@@ -0,0 +1,44 @@
+require "test_helper"
+
+class PlaylistTest < ActiveSupport::TestCase
+  test "invalid if name is blank" do
+    playlist = Playlist.new(name: "", energy: 35, danceability: 56, genre: "genre1")
+    playlist.valid?
+    assert_not playlist.errors[:name].empty?
+  end
+
+  test "invalid if all parameters are blank" do
+    playlist = Playlist.new(name: "name1")
+    playlist.valid?
+    assert_not playlist.errors[:base].empty?
+  end
+
+  test "valid if at least one parameter is filled" do
+    playlist = Playlist.new(name: "name1", energy: 68)
+    playlist.valid?
+    assert_empty playlist.errors[:base]
+  end
+
+  test "valid if danceability & energy are between 1 and 100" do
+    1.upto(100).each do |i|
+      playlist = Playlist.new(name: "name1", danceability: i, energy: i)
+      playlist.valid?
+      assert_empty playlist.errors[:danceability]
+      assert_empty playlist.errors[:energy]
+    end
```

Comment:
en pratique ici c'est plus efficace de mettre une valeur random entre 1 et 100 et vérifier que c'est valide plutôt que de tester chaque valeur: c'est très lent de faire comme tu procèdes (même si c'est une meilleure couverture)

```suggestion
    i = rand(1..100)
    playlist = Playlist.new(name: "name1", danceability: i, energy: i)
    playlist.valid?
    assert_empty playlist.errors[:danceability]
    assert_empty playlist.errors[:energy]
```

Reply:
D'ailleurs ici pour ce test et pour les 2 du dessous il faudrait que tu dédoubles pour `energy` et `danceability` ça serait plus clair (même si c'est plus verbeux)